Automattic/jetpack
# Update videopress shortcode sanitization

## Proposed changes:

* sanitizes the `preload` value provided to a video block or shortcode.
* I *think* this is really just a self-xss but reporting here to be safe.

### Other information:


## Jetpack product discussion



p3btAN-2IZ-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Before applying branch, repro the issue
* Have the Jetpack plugin enabled, but Jetpack VideoPress plugin disabled. VideoPress in the Jetpack settings _should_ be enabled.
* Create a post with the following shortcode: `[wpvideo OcobLTqC freedom=true preloadContent='"src=x onerror=alert(document.cookie) xss']`
* Publish the post then view the post
* You should see the alert with cookie contents
* Apply the branch
* refresh the page
* You should NOT see the alert
* Enable Jetpack VideoPress plugin
* refresh the page
* inspect the video player iframe
* the `src` value should show the default `&preloadContent=metadata` url parameter and not the xss content
